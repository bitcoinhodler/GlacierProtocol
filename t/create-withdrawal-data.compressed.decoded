{
  "txid": "a903f431c3c8c82bbb0cd339b1bc26d340a9dcfe44876bc7b1776e57e5e8b354",
  "hash": "a903f431c3c8c82bbb0cd339b1bc26d340a9dcfe44876bc7b1776e57e5e8b354",
  "version": 2,
  "size": 405,
  "vsize": 405,
  "weight": 1620,
  "locktime": 0,
  "vin": [
    {
      "txid": "88564e99ae78d41fed28d10357ab14f3f7f0ce3d3b38db9ba7a413e71e269d3d",
      "vout": 1,
      "scriptSig": {
        "asm": "0 304402203a3cc0cd87057f2fffc42d3e53efb3fa2c2c743d1c07f6c4db80184901402087022030707a619bd040abf651fdb3538adbed36acae8e2b1f488be38ce7f076d0c4f3[ALL] 30440220241e13b1fbab64c362133132a3eedecba9ff90f399819f44ddc5127a23b129f902202e2b215a978e9a1dab9356f78469b0735b4c157a8e559ddb166e3936b8ffca6f[ALL] 52210277f15f22aeffaf3f3bc48a280a767f7c6af21276783c09ff3bcaabbece178113210203f642223bc751e125ab604ed20c544e933eb82a19201f90bda6236a9ddbedac21026895f3ed01c09025ba9cfab7b67d699e3ebaceada0beff70bc72c7f10df2d1c621039f3cb152130c241e01b9220a147f69f7111b830def35b510973ab47bb9af7ff654ae",
        "hex": "0047304402203a3cc0cd87057f2fffc42d3e53efb3fa2c2c743d1c07f6c4db80184901402087022030707a619bd040abf651fdb3538adbed36acae8e2b1f488be38ce7f076d0c4f3014730440220241e13b1fbab64c362133132a3eedecba9ff90f399819f44ddc5127a23b129f902202e2b215a978e9a1dab9356f78469b0735b4c157a8e559ddb166e3936b8ffca6f014c8b52210277f15f22aeffaf3f3bc48a280a767f7c6af21276783c09ff3bcaabbece178113210203f642223bc751e125ab604ed20c544e933eb82a19201f90bda6236a9ddbedac21026895f3ed01c09025ba9cfab7b67d699e3ebaceada0beff70bc72c7f10df2d1c621039f3cb152130c241e01b9220a147f69f7111b830def35b510973ab47bb9af7ff654ae"
      },
      "sequence": 4294967295
    }
  ],
  "vout": [
    {
      "value": 0.00002380,
      "n": 0,
      "scriptPubKey": {
        "asm": "OP_HASH160 606fb72624603795d73b911a729d88f95f9b6ac8 OP_EQUAL",
        "desc": "addr(2N238gpicqPFfMz8sUN7pqqaJnYkgsC1WmL)#ly84wms3",
        "hex": "a914606fb72624603795d73b911a729d88f95f9b6ac887",
        "address": "2N238gpicqPFfMz8sUN7pqqaJnYkgsC1WmL",
        "type": "scripthash"
      }
    },
    {
      "value": 0.00001000,
      "n": 1,
      "scriptPubKey": {
        "asm": "OP_DUP OP_HASH160 5d1e4870ab9df56432aa357021fd0a15a7094ad1 OP_EQUALVERIFY OP_CHECKSIG",
        "desc": "addr(mp1KQzUoDf9div7jjgon3UuqLThMSo3tmF)#0hm8ddaw",
        "hex": "76a9145d1e4870ab9df56432aa357021fd0a15a7094ad188ac",
        "address": "mp1KQzUoDf9div7jjgon3UuqLThMSo3tmF",
        "type": "pubkeyhash"
      }
    }
  ]
}
