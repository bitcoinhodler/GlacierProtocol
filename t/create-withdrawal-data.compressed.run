#!/bin/bash
set -e

$GLACIERSCRIPT --regtest=$1 create-withdrawal-data << INPUT 2>&1
y
y
y
y
y
y
2N238gpicqPFfMz8sUN7pqqaJnYkgsC1WmL
52210277f15f22aeffaf3f3bc48a280a767f7c6af21276783c09ff3bcaabbece178113210203f642223bc751e125ab604ed20c544e933eb82a19201f90bda6236a9ddbedac21026895f3ed01c09025ba9cfab7b67d699e3ebaceada0beff70bc72c7f10df2d1c621039f3cb152130c241e01b9220a147f69f7111b830def35b510973ab47bb9af7ff654ae
mp1KQzUoDf9div7jjgon3UuqLThMSo3tmF
1
02000000018277cdf95b8c03e54a594ef1114d51672c8bc816f33ca7a7e65950e809779b44010000006a47304402200a60e471be7d1ea4ba969938e6e2aae1bb31839f829b6042a28b05a7e695e6f90220416361938b7ac716b4be37eb5e7a547b7e58c26e12844abba8e91a49a9ae479301210301c32167c39e962bb3cd02a898c0c641425fc960323d0e3372b5b172272d7954ffffffff023d1ffd00000000001976a9141ce7e2c86e0e0e39378e97c97619fb442231cdea88ac881300000000000017a914606fb72624603795d73b911a729d88f95f9b6ac88700000000
2
cSfvoUsWWZ81M4qGYciPHdNxHXQxPEsXBdGWjYdchpisPabtiKVQ
cPWVKeCBjnv833Kayc6nczBbG4YqqbCucup3n5D68WPzBEfceBAQ
1
n
4
y
0.00001000
y
INPUT
