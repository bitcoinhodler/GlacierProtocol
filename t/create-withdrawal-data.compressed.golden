Are you running this on a computer WITHOUT a network connection of any kind? (y/n)?Have the wireless cards in this computer been physically removed? (y/n)?Are you running on battery power? (y/n)?Are you running on an operating system booted from a USB drive? (y/n)?Is your screen hidden from view of windows, cameras, and other people? (y/n)?Are smartphones and all other nearby devices turned off and in a Faraday bag? (y/n)?
You will need to enter several pieces of information to create a withdrawal transaction.


*** PLEASE BE SURE TO ENTER THE CORRECT DESTINATION ADDRESS ***


Source cold storage address: 
Redemption script for source cold storage address: 
Destination address: 
How many unspent transactions will you be using for this withdrawal? 
Please paste raw transaction #1 (hexadecimal format) with unspent outputs at the source address
OR
input a filename located in the current directory which contains the raw transaction data
(If the transaction data is over ~4000 characters long, you _must_ use a file.):

Transaction data found for source address.
TOTAL unspent amount for this raw transaction: 0.00005000 BTC

How many private keys will you be signing this transaction with (at least 2 required to complete transaction)?
#: Key #1: Key #2: 
Enter fee rate.
Satoshis per vbyte: 
Based on the provided rate, the fee will be 0.00000405 bitcoin.
Confirm? (y/n): 
Fee calculation aborted. Starting over...

Enter fee rate.
Satoshis per vbyte: 
Based on the provided rate, the fee will be 0.00001620 bitcoin.
Confirm? (y/n): 
Please enter the decimal amount (in bitcoin) to withdraw to the destination address.

Example: For 2.3 bitcoins, enter "2.3".

After a fee of 0.00001620, you have 0.00003380 bitcoins available to withdraw.

*** Technical note for experienced Bitcoin users:  If the withdrawal amount & fee are cumulatively less than the total amount of the unspent transactions, the remainder will be sent back to the same cold storage address as change. ***

Amount to send to mp1KQzUoDf9div7jjgon3UuqLThMSo3tmF (leave blank to withdraw all funds stored in these unspent transactions): 0.00002380 being returned to cold storage address address 2N238gpicqPFfMz8sUN7pqqaJnYkgsC1WmL.

Is this data correct?
*** WARNING: Incorrect data may lead to loss of funds ***

0.00005000 BTC in unspent inputs from cold storage address 2N238gpicqPFfMz8sUN7pqqaJnYkgsC1WmL
0.00002380 BTC going back to cold storage address 2N238gpicqPFfMz8sUN7pqqaJnYkgsC1WmL
0.00001000 BTC going to destination address mp1KQzUoDf9div7jjgon3UuqLThMSo3tmF
Fee amount: 0.00001620

Signing with private keys: 
cSfvoUsWWZ81M4qGYciPHdNxHXQxPEsXBdGWjYdchpisPabtiKVQ
cPWVKeCBjnv833Kayc6nczBbG4YqqbCucup3n5D68WPzBEfceBAQ


Confirm? (y/n): 
Calculating transaction...

Final fee rate: 4.00 satoshis per vbyte

Raw signed transaction (hex):
02000000013d9d261ee713a4a79bdb383b3dcef0f7f314ab5703d128ed1fd478ae994e568801000000fd1e010047304402203a3cc0cd87057f2fffc42d3e53efb3fa2c2c743d1c07f6c4db80184901402087022030707a619bd040abf651fdb3538adbed36acae8e2b1f488be38ce7f076d0c4f3014730440220241e13b1fbab64c362133132a3eedecba9ff90f399819f44ddc5127a23b129f902202e2b215a978e9a1dab9356f78469b0735b4c157a8e559ddb166e3936b8ffca6f014c8b52210277f15f22aeffaf3f3bc48a280a767f7c6af21276783c09ff3bcaabbece178113210203f642223bc751e125ab604ed20c544e933eb82a19201f90bda6236a9ddbedac21026895f3ed01c09025ba9cfab7b67d699e3ebaceada0beff70bc72c7f10df2d1c621039f3cb152130c241e01b9220a147f69f7111b830def35b510973ab47bb9af7ff654aeffffffff024c0900000000000017a914606fb72624603795d73b911a729d88f95f9b6ac887e8030000000000001976a9145d1e4870ab9df56432aa357021fd0a15a7094ad188ac00000000

Transaction fingerprint (md5):
1273ceacffacd56e44474ce536c97d1d
QR code for transaction written to transaction.png
